src/distilabel/steps/tasks/argilla_labeller.py

```diff
@@ -0,0 +1,614 @@
+                LabelQuestion,
+                RatingQuestion,
+                SpanQuestion,
+                TextQuestion,
+            ],
+        ],
+    ) -> ChatType:
+        """Format the input into a chat message.
+
+        Args:
+            input (Dict[str, Union[Dict[str, Any], Record, TextField, MultiLabelQuestion, LabelQuestion, RatingQuestion, SpanQuestion, TextQuestion]]): The input to format.
+
+        Returns:
+            ChatType: The formatted chat message.
+        """
+        record = input[self.inputs[0]]
+        fields = input.get(self.optional_inputs[0], self.fields)
+        question = input.get(self.optional_inputs[1], self.question)
+        examples = input.get(self.optional_inputs[2], self.example_records)
+        guidelines = input.get(self.optional_inputs[3], self.guidelines)
+
+        if any([fields is None, question is None]):
+            raise ValueError(
+                "Fields and question must be provided during init or through `process` method."
+            )
+
+        record = record.to_dict() if not isinstance(record, dict) else record
+        question = question.serialize() if not isinstance(question, dict) else question
+        fields = [
+            field.serialize() if not isinstance(field, dict) else field
+            for field in fields
+        ]
+        examples = (
+            [
+                example.to_dict() if not isinstance(example, dict) else example
+                for example in examples
+            ]
+            if examples
+            else None
+        )
+
+        formatted_fields = self._format_record(record, fields)
+        formatted_question = self._format_question(question)
+        formatted_examples = (
+            self._format_example_records(examples, fields, question)
+            if examples
+            else False
+        )
+        prompt = self._template.render(
+            fields=formatted_fields,
+            question=formatted_question,
+            examples=formatted_examples,
+            guidelines=guidelines,
+        )
+
+        messages = []
+        if self.system_prompt:
+            messages.append({"role": "system", "content": self.system_prompt})
+        messages.append({"role": "user", "content": prompt})
+        return messages
+
+    @property
+    def outputs(self) -> List[str]:
+        return ["suggestion"]
+
+    def format_output(
+        self, output: Union[str, None], input: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Format the output into a dictionary.
+
+        Args:
+            output (Union[str, None]): The output to format.
+            input (Dict[str, Any]): The input to format.
+
+        Returns:
+            Dict[str, Any]: The formatted output.
+        """
+        question: (
+            Any
+            | Dict[str, Any]
+            | LabelQuestion
+            | MultiLabelQuestion
+            | RatingQuestion
+            | TextQuestion
+            | None
```

Comment:
We should `Union` instead of the new syntax as it won't work otherwise for Python 3.9